File: src/faction.cpp
/******************************************************************************
 *	Title:		Faction
 *	Authour:	Alistair Hudson
 *	Reviewer:	
 *	Version:	20.08.2020.0
 ******************************************************************************/
#include "faction.hpp"

using namespace the_prince;

/******MACROS******/
#define MAX_ATTITUDE   (100)
#define MAX_INFLUENCE  (100)

/******TYPEDEFS*****/

/****** GLOBAL VARIABLES*****/

/*****STRUCTS*******/

/*****CLASSES******/

/*****FUNCTORS*****/

/******INTERNAL FUNCTION DECLARATION******/


/******CLASS METHODS*******/
/*===Facton===*/
void Faction::DecreaseAttitudeLevel(float dcr)
{
    m_attitude -= dcr;
    if (m_attitude <= -MAX_ATTITUDE)
    {
        m_attitude = -MAX_ATTITUDE;
    }
    if (m_attitude >= MAX_ATTITUDE)
    {
        m_attitude = MAX_ATTITUDE;
    }
}

void Faction::IncreaseAttitudeLevel(float inc)
{
    m_attitude += inc;
    if (m_attitude >= MAX_ATTITUDE)
    {
        m_attitude = MAX_ATTITUDE;
    }
    if (m_attitude <= -MAX_ATTITUDE)
    {
        m_attitude = -MAX_ATTITUDE;
    }
}

void Faction::DecreaseInfluenceLevel(int dcr)
{
    m_influence -= dcr;
    if (m_influence >= MAX_INFLUENCE)
    {
        m_influence = MAX_INFLUENCE;
    }
    if (m_influence <= 0)
    {
        m_influence = 0;
    }
}

void Faction::IncreaseInfluenceLevel(int inc)
{
    m_influence += inc;
    if (m_influence >= MAX_INFLUENCE)
    {
        m_influence = MAX_INFLUENCE;
    }
    if (m_influence <= 0)
    {
        m_influence = 0;
    }
}


/*****FUNCTION DEFINITION******/
 

File: Unity/The Prince/Assets/Scripts/Faction.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Faction : MonoBehaviour
{
    //Variables
    float m_attitude = 0;
    float m_influence = 0;

    //Constants
    float MAX_ATTITUDE = 100;
    float MAX_INFLUENCE = 100;

    public void DecreaseAttitudeLevel(float dcr)
    {
        m_attitude -= dcr;
        if (m_attitude <= -MAX_ATTITUDE)
        {
            m_attitude = -MAX_ATTITUDE;
        }
        if (m_attitude >= MAX_ATTITUDE)
        {
            m_attitude = MAX_ATTITUDE;
        }
    }

    public void IncreaseAttitudeLevel(float inc)
    {
        m_attitude += inc;
        if (m_attitude >= MAX_ATTITUDE)
        {
            m_attitude = MAX_ATTITUDE;
        }
        if (m_attitude <= -MAX_ATTITUDE)
        {
            m_attitude = -MAX_ATTITUDE;
        }
    }

    public void DecreaseInfluenceLevel(int dcr)
    {
        m_influence -= dcr;
        if (m_influence >= MAX_INFLUENCE)
        {
            m_influence = MAX_INFLUENCE;
        }
        if (m_influence <= 0)
        {
            m_influence = 0;
        }
    }

    public void IncreaseInfluenceLevel(int inc)
    {
        m_influence += inc;
        if (m_influence >= MAX_INFLUENCE)
        {
            m_influence = MAX_INFLUENCE;
        }
        if (m_influence <= 0)
        {
            m_influence = 0;
        }
    }

    public float GetAttitude()
    {
        return m_attitude;
    }

    public float GetInfluence()
    {
        return m_influence;
    }

    public bool Rebelion()
    {
        if (Random.Range(0, 100) <= Mathf.Abs(m_attitude) * (float)m_influence / 100)
        {
            return true;
        }
        return false;
    }
}
